corehq/apps/hqwebapp/models.py

```diff
@@ -295,8 +295,11 @@ def sidebar_items(self):
         ])]
 
         user_reports = []
-        if (toggle_enabled(self._request, toggles.REPORT_BUILDER)
-                and has_privilege(self._request, privileges.REPORT_BUILDER)):
+
+        builder_enabled = toggle_enabled(self._request, toggles.REPORT_BUILDER)
+        builder_privileges = has_privilege(self._request, privileges.REPORT_BUILDER)
+        beta_group_enabled = toggle_enabled(self._request, toggles.REPORT_BUILDER_BETA_GROUP)
+        if ((builder_enabled and builder_privileges) or beta_group_enabled):
```

Comment:
would be nice if the logic here was moved to its own function to reduce code duplication